css-display-4/Overview.bs

```diff
@@ -1176,6 +1188,39 @@ Reading Order: the 'reading-order' property</h3>
 
 	ISSUE: Is this the most appropriate interaction with with <code>tabindex</code>?
 
+<h4 id="randomizing-layout">
+Randomizing Layout Methods and ''reading-order: auto''</h4>
+
+	<dfn>Randomizing formatting contexts</dfn>
+	are formatting contexts in which which can indiscriminately scramble the layout order,
+	as listed below.
+	Since use of such methods indicates that the underlying order of the elements
+	is not important,
+	the UA will match the [=reading and navigation order=] of the elements
+	to match the layout order of their boxes as follows:
+
+	<dl>
+		<dt>''grid-auto-flow: dense''
+		<dd>
+			When ''grid-auto-flow: dense'' <a spec=css-grid-1>automatic placement</a>
+			changes the relative order of an item
+			with respect to other items with ''reading-order/auto'' or ''reading-order/0'' 'reading-order'
+			by shifting it earlier in the grid placement order,
+			then its placement in the [=reading and navigation order=] is likewise shifted.
+
+			<div class="example">
+				For example, [=grid items=] are laid out in a ''grid-auto-flow: column dense'' grid,
+				they are first re-ordered to [=order-modified document order=]
+				before applying the [[css-grid-1#auto-placement-algo]].
+				Among the subset of these items whose ''reading-order'' is ''0'' or ''auto'',
+				those with ''reading-order: auto'' are shifted
+				to the latest position in the [=reading and navigation order=]
+				that is also after any items with ''reading-order: 0'' or ''reading-order: auto''
+				that came earlier in the [=grid-modified document order=]
```

Comment:
What's the grid-modified document order? Grid is 2D, so I don't really understand how to map it to a 1D sequence.

Reply:
This is a defined term in the Grid spec.